src/test/java/spoon/MavenLauncherTest.java

```diff
@@ -42,39 +42,36 @@ public void scan(CtElement element) {
 
 	@Test
 	public void spoonMavenLauncherTest() {
+		// contract: MavenLauncher finds the right number of sources in all three modes
+		String testProject = "./src/test/resources/maven-launcher/pac4j";
+
 		// without the tests
-		MavenLauncher launcher = new MavenLauncher("./", MavenLauncher.SOURCE_TYPE.APP_SOURCE);
+		MavenLauncher launcher = new MavenLauncher(testProject, MavenLauncher.SOURCE_TYPE.APP_SOURCE);
 
-		assertEquals(26, launcher.getEnvironment().getSourceClasspath().length);
+		// only with APP_SOURCE
+		assertEquals(99, launcher.getEnvironment().getSourceClasspath().length);
 
-		// 56 because of the sub folders of src/main/java
-		assertEquals(59, launcher.getModelBuilder().getInputSources().size());
 
-		// with the tests
-		launcher = new MavenLauncher("./", MavenLauncher.SOURCE_TYPE.ALL_SOURCE);
+		// number of the sub folders of src/main/java
+		assertEquals(0, launcher.getModelBuilder().getInputSources().size());
 
-		assertEquals(33, launcher.getEnvironment().getSourceClasspath().length);
+		// with the tests
+		launcher = new MavenLauncher(testProject, MavenLauncher.SOURCE_TYPE.ALL_SOURCE);
 
-		// number of the sub folders of src/main/java and src/test/java
-		assertTrue("size: " + launcher.getModelBuilder().getInputSources().size(), launcher.getModelBuilder().getInputSources().size() >= 220);
+		assertEquals(8, launcher.getEnvironment().getComplianceLevel());
+		assertEquals(0, launcher.getModelBuilder().getInputSources().size());
+		// more with ALL_SOURCE
+		assertEquals(166, launcher.getEnvironment().getSourceClasspath().length);
 
 		// specify the pom.xml
-		launcher = new MavenLauncher("./pom.xml", MavenLauncher.SOURCE_TYPE.APP_SOURCE);
+		launcher = new MavenLauncher(testProject, MavenLauncher.SOURCE_TYPE.APP_SOURCE);
```

Comment:
this line looks same like line 49 above. 
A) delete it
B) let it use pom.xml as says contract of the test